genesyscloud/external_contacts_organization/genesyscloud_externalcontacts_organization_proxy.go

```diff
@@ -0,0 +1,201 @@
+	for _, externalOrganization := range allExternalOrganizations {
+		if externalOrganization.Id == nil {
+			continue
+		}
+		rc.SetCache(p.externalOrganizationCache, *externalOrganization.Id, externalOrganization)
+	}
+
+	return &allExternalOrganizations, response, nil
+}
+
+// getExternalContactsOrganizationIdByNameFn is an implementation of the function to get a Genesys Cloud external contacts organization by name
+func getExternalContactsOrganizationIdByNameFn(ctx context.Context, p *externalContactsOrganizationProxy, name string) (id string, retryable bool, apiResponse *platformclientv2.APIResponse, err error) {
+
+	externalOrganizations, response, err := getAllExternalContactsOrganizationFn(ctx, p, name)
+	if err != nil {
+		return "", false, response, err
+	}
+
+	if externalOrganizations == nil || len(*externalOrganizations) == 0 {
+		return "", true, response, fmt.Errorf("no external contacts organization found with name %s", name)
+	}
+
+	var externalOrganization platformclientv2.Externalorganization
+	for _, externalOrganizationSdk := range *externalOrganizations {
+		if *externalOrganization.Name == name {
+			log.Printf("Retrieved the external contacts organization id %s by name %s", *externalOrganizationSdk.Id, name)
+			externalOrganization = externalOrganizationSdk
+			return *externalOrganization.Id, false, response, nil
+		}
+	}
+
+	return "", false, response, fmt.Errorf("unable to find external contacts organization with name %s", name)
+}
+
+// getExternalContactsOrganizationByIdFn is an implementation of the function to get a Genesys Cloud external contacts organization by Id
+func getExternalContactsOrganizationByIdFn(ctx context.Context, p *externalContactsOrganizationProxy, id string) (externalContactsOrganization *platformclientv2.Externalorganization, apiResponse *platformclientv2.APIResponse, err error) {
+	if externalOrganization := rc.GetCacheItem(p.externalOrganizationCache, id); externalOrganization != nil {
+		return externalOrganization, nil, nil
+	}
+	externalOrganization, response, err := p.externalContactsApi.GetExternalcontactsOrganization(id, []string{}, false)
+	if err != nil {
+		return nil, response, fmt.Errorf("failed to retrieve external contacts organization by id %s: %s", id, err)
+	}
+
+	return externalOrganization, response, nil
+}
+
+// updateExternalContactsOrganizationFn is an implementation of the function to update a Genesys Cloud external contacts organization
+func updateExternalContactsOrganizationFn(ctx context.Context, p *externalContactsOrganizationProxy, id string, externalContactsOrganization *platformclientv2.Externalorganization) (*platformclientv2.Externalorganization, *platformclientv2.APIResponse, error) {
+	externalOrganization, response, err := p.externalContactsApi.PutExternalcontactsOrganization(id, *externalContactsOrganization)
+	if err != nil {
+		return nil, response, fmt.Errorf("failed to update external contacts organization: %s", err)
```

Comment:
There is no need to add extra context to the error here since we will do that in the invoking functions. It can just be one line:

```go
return p.externalContactsApi.PutExternalcontactsOrganization(id, *externalContactsOrganization)
```


 Same with the other functions in this file (create, delete, getById, etc.)